examples/digital_fingerprinting/production/morpheus/benchmarks/test_bench_e2e_dfp_pipeline.py

```diff
@@ -34,334 +33,467 @@
 
-    dfp_config = DFPConfig(pipeline_conf, feature_columns, source="duo")
+@pytest.mark.benchmark
+def test_dfp_stages_duo_inference_e2e(benchmark: typing.Any, tmp_path):
 
-    config = dfp_config.get_config()
-    stages_conf = dfp_config.get_stages_conf()
-    filenames = dfp_config.get_filenames()
+    pipe_conf = PIPELINES_CONF.get("test_dfp_stages_duo_inference_e2e")
 
-    source_schema = get_duo_source_schema(config)
-    preprocess_schema = get_duo_preprocess_schema(config)
+    bcg = BenchmarkConfGenerator(pipe_conf=pipe_conf)
+
+    pipe_config = bcg.pipe_config
+    stages_conf = bcg.get_stages_conf()
+    input_filenames = bcg.get_filenames()
+    schema: Schema = bcg.get_schema()
+
+    output_filepath = os.path.join(tmp_path, "detections_duo.csv")
 
-    benchmark(dfp_training_pipeline_stages, config, stages_conf, source_schema, preprocess_schema, filenames)
+    benchmark(dfp_inference_pipeline_stages,
+              pipe_config,
+              stages_conf,
+              schema.source,
+              schema.preprocess,
+              input_filenames,
+              output_filepath)
 
 
 @pytest.mark.benchmark
-def test_dfp_training_azure_modules_e2e(benchmark: typing.Any):
+def test_dfp_modules_azure_payload_inference_e2e(benchmark: typing.Any):
+
+    pipe_conf = PIPELINES_CONF.get("test_dfp_modules_azure_payload_inference_e2e")
 
-    feature_columns = [
-        "appDisplayName",
-        "clientAppUsed",
-        "deviceDetailbrowser",
-        "deviceDetaildisplayName",
-        "deviceDetailoperatingSystem",
-        "statusfailureReason",
-        "appincrement",
-        "locincrement",
-        "logcount"
-    ]
+    bcg = BenchmarkConfGenerator(pipe_conf=pipe_conf)
 
-    pipeline_conf = PIPELINES_CONF.get("test_dfp_training_azure_modules_e2e")
+    pipe_config = bcg.pipe_config
+    module_config = bcg.get_module_conf()
+    input_filenames = bcg.get_filenames()
 
-    dfp_config = DFPConfig(pipeline_conf, feature_columns, source="azure", modules_conf=MODULES_CONF)
+    benchmark(dfp_modules_pipeline, pipe_config, module_config, filenames=input_filenames)
+
+
+@pytest.mark.benchmark
+def test_dfp_modules_azure_payload_lti_e2e(benchmark: typing.Any):
 
-    config = dfp_config.get_config()
-    filenames = dfp_config.get_filenames()
+    pipe_conf = PIPELINES_CONF.get("test_dfp_modules_azure_payload_lti_e2e")
 
-    source_schema = get_azure_source_schema(config)
-    preprocess_schema = get_azure_preprocess_schema(config)
+    bcg = BenchmarkConfGenerator(pipe_conf=pipe_conf)
 
-    dfp_config.update_modules_conf(source_schema, preprocess_schema)
+    pipe_config = bcg.pipe_config
+    module_config = bcg.get_module_conf()
+    input_filenames = bcg.get_filenames()
 
-    benchmark(dfp_training_pipeline_modules, config, dfp_config.modules_conf, filenames)
+    benchmark(dfp_modules_pipeline, pipe_config, module_config, filenames=input_filenames)
 
 
+@pytest.mark.skip_by_default(
```

Comment:
Lets try to use the default functionality in pytest for skipping on missed dependencies: https://docs.pytest.org/en/7.1.x/how-to/skipping.html#skipping-on-a-missing-import-dependency